app/models/shipit/user.rb

```diff
@@ -123,5 +129,20 @@ def identify_renamed_user!
     rescue Octokit::NotFound
       false
     end
+
+    def org_email?(email_address)
+      email_address.present? && email_address.split("@").last&.include?(org_domain)
+    end
+
+    def filter_org_emails(emails)
+      emails
+        .select { |email| org_email?(email.email) }
+        .map(&:email)
+    end
+
+    # Domain to filter email addresses by. Override in integration application.
+    def org_domain
+      ""
```

Comment:
I don't like this way of configuring it. I'd rather see a property on the `Shipit` namespace. There is plenty of those already.

Reply:
Would that best be done as a class variable on User (or some other class?), a module variable in a new file (itself just `def`ing Shipit), or as an overridden constant in one of the above?

I'd be interested in an example if you have one, as I'm not sure if the things I'm looking at are what you're talking about. I'm still a bit green on the way we do things with Ruby and Rails.

Reply:
No, `Shipit` is better, as it's the app namespace, to it make sense to centralize this kind of configurations there.

As for the example, there is some here: https://github.com/Shopify/shipit-engine/blob/6bd83a31ad88ed81bc920d9d57c83055110bb815/lib/shipit.rb#L62

Reply:
Ah, thank you. This has led to some more questions though:

I noticed in testing that timeout_exit_codes as set below is actually nil at runtime.
```ruby
 attr_accessor :disable_api_authentication, :timeout_exit_codes, :preferred_org_emails
 self.timeout_exit_codes = [].freeze
// calling Shipit.timeout_exit_codes anywhere in the code returns nil, rather than an empty array.
```
Do you know if that's intended? `@timeout_exit_codes = [].freeze` also behaves the same, but `@@timeout_exit_codes = [].freeze` results in the var defaulting to an empty array at runtime.

I'm not sure if I'm just misunderstanding Rubys' runtime class structure or something like that.
It's overridden in the Shipit project and relevant unit test, so it doesn't appear to be an issue, but I think the intent here was to have a non-nil default value?

Reply:
Are you sure? I just checked and it works fine:

```
>> Shipit.timeout_exit_codes
=> []
>> Shipit.timeout_exit_codes.frozen?
=> true
```

If you get it `nil` it might be that something overwrites it.

Reply:
Ah, this may be some test specific thing.
Rails console returns an empty and frozen array, but calling `Shipit.timeout_exit_codes` and similar from Test entry points (either directly, or via a class like User making the call) results in nil.

I thought I had observed it while :tophat: testing the email logic, but perhaps I was mistaken.